inference/inference-core/src/commonMain/kotlin/io/kinference.core/operators/math/BiasGelu.kt

```diff
@@ -43,33 +47,89 @@ class BiasGeluVer1(name: String, attributes: Map<String, Attribute<Any>> = empty
 
         private val SQRT_2_1_FLOAT = 1f / sqrt(2f)
         private val SQRT_2_1 = 1.0 / sqrt(2.0)
-    }
 
+        private fun computeFloat(output: FloatNDArray, input: FloatNDArray, bias: FloatNDArray, startOffset: Int = 0, count: Int = input.linearSize - startOffset) {
+            val inputPointer = input.array.pointer(startOffset)
+            val outputPointer = output.array.pointer(startOffset)
+            val biasPointer = bias.array.pointer()
+
+            outputPointer.acceptWithRecursive(inputPointer, biasPointer, count) { _: Float, input: Float, bias: Float ->
+                val value = input + bias
+
+                val valueForErf = value * SQRT_2_1_FLOAT
+                val sign = valueForErf.sign
+                val absValue = abs(valueForErf)
+                val t = 1f / (1f + ERF_P_VALUE_FLOAT * absValue)
+                val sum = t * (ERF_COEF_1_FLOAT + t * (ERF_COEF_2_FLOAT + t * (ERF_COEF_3_FLOAT + t * (ERF_COEF_4_FLOAT + t * ERF_COEF_5_FLOAT))))
+                val erfValue = sign * (1f - sum * exp(-absValue * absValue))
+                0.5f * value * (1.0f + erfValue)
+            }
+        }
+
+        private fun computeDouble(output: DoubleNDArray, input: DoubleNDArray, bias: DoubleNDArray, startOffset: Int = 0, count: Int = input.linearSize - startOffset) {
+            val inputPointer = input.array.pointer(startOffset)
+            val outputPointer = output.array.pointer(startOffset)
+            val biasPointer = bias.array.pointer()
+
+            outputPointer.acceptWithRecursive(inputPointer, biasPointer, count) { _: Double, input: Double, bias: Double ->
+                val value = input + bias
+
+                val valueForErf = value * SQRT_2_1
+                val sign = valueForErf.sign
+                val absValue = abs(valueForErf)
+                val t = 1.0 / (1.0 + ERF_P_VALUE * absValue)
+                val sum = t * (ERF_COEF_1 + t * (ERF_COEF_2 + t * (ERF_COEF_3 + t * (ERF_COEF_4 + t * ERF_COEF_5))))
+                val erfValue = sign * (1.0 - sum * exp(-absValue * absValue))
+
+                0.5 * value * (1.0 + erfValue)
+            }
+        }
+
+        private fun <T : NumberNDArrayCore> computeBatched(
+            coroutineContext: CoroutineContext,
+            input: T,
+            output: T,
+            bias: T,
+            batchFunc: (T, T, T, Int, Int) -> Unit
+        ) {
+            val rowSize = bias.linearSize
+            val numThreads = PlatformUtils.availableThreads
+            val numRows = input.linearSize / bias.linearSize
+            val numBatches = if (numRows < numThreads) numRows else numThreads
+            val batchSize = floor(numRows.toDouble() / numBatches).toInt()
+            val endBatchOffsets = IntArray(numBatches) { batchSize * (it + 1) * rowSize }.apply {
```

Comment:
Do you need that array at all? It seems you can generate it on the go :) Just in for